code/modules/clothing/head/misc_special.dm

```diff
@@ -139,16 +139,96 @@
 /*
  * Pumpkin head
  */
-/obj/item/clothing/head/pumpkinhead
+
+/obj/item/clothing/head/pumpkin
 	name = "carved pumpkin"
-	desc = "A jack o' lantern! Believed to ward off evil spirits."
-	icon_state = "hardhat0_pumpkin"//Could stand to be renamed
+	desc = "A pumpkin with a spooky face carved on it. Looks like it needs a candle."
+	icon_state = "pumpkin_carved"
 	flags_inv = HIDEMASK|HIDEEARS|HIDEEYES|HIDEFACE|BLOCKHAIR
 	body_parts_covered = HEAD|FACE|EYES
-	brightness_on = 2
-	light_overlay = "helmet_light"
-	w_class = 3
 	drop_sound = 'sound/items/drop/herb.ogg'
+	w_class = 3
+	throwforce = 1
+	throw_speed = 0.5
+
+/obj/item/clothing/head/pumpkin/attackby(var/obj/O, mob/user as mob)
+    if(istype(O, /obj/item/weapon/flame/candle))
+        var/obj/item/weapon/flame/candle/c = O
+        var/candle_wax = c.wax
+        to_chat(user, "You add [O] to [src].")
+        playsound(src.loc, 'sound/items/drop/gloves.ogg', 50, 1)
+        qdel(O)
+        var/obj/item/clothing/head/pumpkin/lantern/L = new /obj/item/clothing/head/pumpkin/lantern(user.loc)
+        L.wax = candle_wax
+        user.put_in_hands(L)
+        qdel(src)
+        return
+
+/obj/item/clothing/head/pumpkin/lantern
+	name = "jack o' lantern"
+	desc = "A pumpkin with a spooky face carved on it, with a candle inside. Believed to ward off evil spirits."
+	light_color = "#E09D37"
+	var/wax = 900
+	var/lit = FALSE
+
+/obj/item/clothing/head/pumpkin/lantern/update_icon()
+	icon_state = "pumpkin_carved[lit ? "_lit" : ""]"
+	if(ismob(loc))
+		var/mob/living/M = loc
+		M.update_inv_head(0)
+		M.update_inv_l_hand(0)
+		M.update_inv_r_hand(1)
+
+/obj/item/clothing/head/pumpkin/lantern/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	..()
+	if(W.iswelder())
+		var/obj/item/weapon/weldingtool/WT = W
+		if(WT.isOn()) //Badasses dont get blinded by lighting their candle with a welding tool
+			light("<span class='notice'>\The [user] casually lights the [name] with [W].</span>")
+	else if(isflamesource(W))
+		light()
+	else if(istype(W, /obj/item/weapon/flame/candle))
+		var/obj/item/weapon/flame/candle/C = W
+		if(C.lit)
+			light()
+
+/obj/item/clothing/head/pumpkin/lantern/proc/light(var/flavor_text = "<span class='notice'>\The [usr] lights the [name].</span>")
+	if(!src.lit)
+		src.lit = TRUE
+		playsound(src.loc, 'sound/items/cigs_lighters/cig_light.ogg', 50, 1)
+		//src.damtype = "fire"
+		for(var/mob/O in viewers(usr, null))
+			O.show_message(flavor_text, 1)
+		set_light(CANDLE_LUM)
+		update_icon()
+		update_clothing_icon()
+		START_PROCESSING(SSprocessing, src)
+
+/obj/item/clothing/head/pumpkin/lantern/process(mob/user as mob)

```

Comment:
```suggestion
/obj/item/clothing/head/pumpkin/lantern/process(mob/user)
```

Same shit, who is supposed to be user here

Reply:
so people can see in chat if a candle snuffed out on their screen